lib/pbench/server/cache_manager.py

```diff
@@ -1133,27 +1134,46 @@ def get_results(self, lock: LockManager) -> Path:
                 uend - ustart,
             )
 
-        self.last_ref.touch(exist_ok=True)
-
-        # If we have a Dataset, and haven't already done this, compute the
-        # unpacked size and record it in metadata so we can use it later.
-        ssize = time.time()
-        if self.dataset and not Metadata.getvalue(
-            self.dataset, Metadata.SERVER_UNPACKED
-        ):
-            try:
-                process = subprocess.run(
-                    ["du", "-s", "-B1", str(self.unpacked)],
-                    capture_output=True,
-                    text=True,
+            # If we have a Dataset, and haven't already done this, compute the
+            # unpacked size and record it in metadata so we can use it later.
+            ssize = time.time()
+            if self.dataset:
+                if not Metadata.getvalue(self.dataset, Metadata.SERVER_UNPACKED):
+                    try:
+                        process = subprocess.run(
+                            ["du", "-s", "-B1", str(self.unpacked)],
+                            capture_output=True,
+                            text=True,
+                        )
+                        if process.returncode == 0:
+                            size = int(process.stdout.split("\t", maxsplit=1)[0])
+                            self.unpacked_size = size
+                            Metadata.setvalue(
+                                self.dataset, Metadata.SERVER_UNPACKED, size
+                            )
+                    except Exception as e:
+                        self.logger.warning("usage check failed: {}", e)
+
+                # Update the time-to-unpack statistic. If the metadata doesn't exist,
+                # or somehow isn't a dict (JSONOBJECT), create a new metric: otherwise
+                # update the existing metric.
+                unpack_time = uend - ustart
+                metric: JSONOBJECT = Metadata.getvalue(
+                    self.dataset, Metadata.SERVER_UNPACK_PERF
                 )
-                if process.returncode == 0:
-                    size = int(process.stdout.split("\t", maxsplit=1)[0])
-                    self.unpacked_size = size
-                    Metadata.setvalue(self.dataset, Metadata.SERVER_UNPACKED, size)
-            except Exception as e:
-                self.logger.warning("usage check failed: {}", e)
-        self.logger.info("{}: size update {:.3f}", self.name, time.time() - ssize)
+                if not isinstance(metric, dict):
+                    metric = {"min": unpack_time, "max": unpack_time, "count": 1}
+                else:
+                    # We max out at about 12 hours here, which seems safely excessive!
+                    metric["min"] = min(metric.get("min", 1000000.0), uend - ustart)
+                    metric["max"] = max(metric.get("max", 0), uend - ustart)
```

Comment:
Do you really want to be re-calculating `uend - ustart` in these two expressions?...or should we be using the value in `unpack_time` which we set at line 1160?

Reply:
That's an oversight: I guess it's just whether I take time to open the branch again and push another commit. Not today, but then there's no big rush either.

Reply:
Nope; no rush.